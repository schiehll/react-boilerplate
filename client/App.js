import React, { Fragment } from 'react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import GlobalStyles from 'styles/global'
import Routes from 'routes'

const App = () => (
  <ThemeProvider theme={theme}>
    <Fragment>
      <GlobalStyles />
      <Routes />
    </Fragment>
  </ThemeProvider>
)

export default App
