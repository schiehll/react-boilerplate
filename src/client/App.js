import React, { Component, Fragment } from 'react'
import { ThemeProvider } from 'styled-components'
import theme from 'client/styles/theme'
import GlobalStyles from 'client/styles/global'
import Routes from 'client/routes'

class App extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <Fragment>
          <GlobalStyles />
          <Routes />
        </Fragment>
      </ThemeProvider>
    )
  }
}

export default App
