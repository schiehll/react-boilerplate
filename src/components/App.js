import React from 'react'
import Link from 'react-router/lib/Link'

const App = ({children}) => {
  return (
    <div>
      <nav>
        <Link to='/'>Home</Link>
        <Link to='/about'>About</Link>
      </nav>
      <main>
        {children}
      </main>
    </div>
  )
}

export default App
