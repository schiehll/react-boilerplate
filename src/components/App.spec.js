import React from 'react'
import {render} from 'enzyme'
import {renderToJson} from 'enzyme-to-json'
import App from './App'

test('should render just fine', () => {
  const component = render(<App />)

  expect(renderToJson(component)).toMatchSnapshot()
})
