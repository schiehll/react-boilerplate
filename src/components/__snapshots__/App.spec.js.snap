exports[`test should render just fine 1`] = `
<div>
  <nav>
    <a>
      Home
    </a>
    <a>
      About
    </a>
  </nav>
  <main />
</div>
`;
